lib/ae_mdw/channels.ex

```diff
@@ -175,4 +177,34 @@ defmodule AeMdw.Channels do
   end
 
   defp deserialize_scope(_nil_or_txis_scope), do: nil
+
+  # Gets from the oldest block state tree since some channels might be absent from newer blocks
+  defp get_oldest_block_hash(_state, nil, update_block_hash, _update_block_index),
+    do: update_block_hash
+
+  defp get_oldest_block_hash(state, param_block_hash, update_block_hash, update_block_index) do
+    case get_block_index(state, param_block_hash) do
+      {:ok, height, mbi} ->
+        if update_block_index < {height, mbi} do
+          update_block_hash
+        else
+          param_block_hash
+        end
+
+      {:error, _reason} ->
+        update_block_hash
+    end
+  end
+
+  defp get_block_index(state, param_block_hash) do
+    block_hash = Validate.id!(param_block_hash)
```

Comment:
Perhaps we can send the decoded block_hash binary to avoid decoding it twice

Reply:
It was avoiding an additional parameter since the type is also needed and it's a few microseconds operation but did the optimization though.